lib/prelude.dx

```diff
@@ -2469,3 +2529,6 @@ instance {a b} [Ix a, Ix b] Ix (a=>b)
   size = intpow (size b) (size a)
   ordinal             = reversed_digits_to_int
   unsafe_from_ordinal = int_to_reversed_digits
+
+instance {a b} [Ix a, NonEmpty b] NonEmpty (a=>b)
```

Comment:
I don't think that `b` needs to be non-empty. `=>` is an integer exponential for which `0^0 = 1` (an empty list is there, even for a `Void` element type)

Reply:
Ah, I agree that `0^0 = 1`, and that this possibility isn't covered by my current rule.  Maybe I'll add an instance `instance {a b} [Ix a, Ix b] NonEmpty (Fin 0) (Fin 0)`.

But `0^1 = 0`, i.e. we can't construct an element of type `Unit=>(Fin 0)`.  So to avoid that case, we can't have an unconstrained rule like `instance {a b} [Ix a, Ix b] NonEmpty a b`

Reply:
Ah, good point! I think the current rule is sufficient